dot/parachain/overseer/overseer.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package overseer
+
+import "fmt"
+
+type Overseer struct {
+	subsystems        map[Subsystem]*Context
+	subsystemMessages map[Subsystem]chan any
+}
+
+type ExamlpeSender struct {
+	senderChan chan any
+}
+
+func (s *ExamlpeSender) SendMessage(msg any) error {
+	s.senderChan <- msg
+	return nil
+}
+
+func NewOverseer() *Overseer {
+	return &Overseer{
+		subsystems:        make(map[Subsystem]*Context),
+		subsystemMessages: make(map[Subsystem]chan any),
+	}
+}
+
+func (o *Overseer) RegisterSubsystem(subsystem Subsystem) {
+	// TODO: determine best buffer size
+	comChan := make(chan any, 5)
+	o.subsystemMessages[subsystem] = comChan
+	receiverChan := make(chan any, 5)
+	o.subsystems[subsystem] = &Context{
+		Sender:   &ExamlpeSender{senderChan: comChan},
+		Receiver: receiverChan,
+	}
+}
+
+func (o *Overseer) Start() {
+	// start subsystems
+	for subsystem, context := range o.subsystems {
+		go func(sub Subsystem, ctx *Context) {
+			err := sub.Run(ctx)
+			if err != nil {
+				// TODO: handle error
+				fmt.Printf("error running subsystem %v: %v\n", sub, err)
```

Comment:
We should add a logger on top the file and use that here to log error, sorta like
```
var (
	logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-overseer"))
)
```